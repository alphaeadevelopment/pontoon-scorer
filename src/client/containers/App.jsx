import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import injectSheet from 'react-jss';
import Footer from './Footer';
import Header from './Header';
import Body from './Body';
import { isDrawerOpen } from '../selectors/drawerSelectors';
import { closeDrawer } from '../actions';
import { drawerRef } from '../lib/drawer';

const styles = {
  root: {
    position: 'relative',
  },
  headerContainer: {

  },
  footerContainer: {
    position: 'fixed',
    width: '100%',
    top: '100%',
    transform: 'translateY(-100%) translateZ(0)',
  },
  bodyContainer: {

  },
};
@connect(state => (
  {
    drawerOpen: isDrawerOpen(state),
  }
), {
  closeDrawer,
})
@injectSheet(styles)
class App extends React.Component {
  state = {
    footerHeight: 0,
  }
  setFooterHeight = (h) => {
    this.setState({ footerHeight: h });
  }
  render() {
    const { classes } = this.props;
    return (
      <div className={classes.root}>
        <div id='drawer-root' ref={drawerRef} />
        <div className={classes.headerContainer}>
          <Header />
        </div>
        <div className={classes.bodyContainer} style={{ marginBottom: this.state.footerHeight }}>
          <Body />
        </div>
        <div className={classes.footerContainer}>
          <Footer onSetHeight={this.setFooterHeight} />
        </div>
      </div>
    );
  }
}

export default App;
