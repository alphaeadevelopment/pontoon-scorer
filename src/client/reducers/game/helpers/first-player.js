import findIndex from 'lodash/findIndex';

export default ({ players, dealer }) => findIndex(players, p => p.idx !== dealer);
