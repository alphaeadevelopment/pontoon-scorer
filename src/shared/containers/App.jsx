import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import injectSheet from 'react-jss';
import Footer from './Footer';
import Header from './Header';
import Body from './Body';
import { isDrawerOpen } from '../selectors/drawerSelectors';
import { closeDrawer, gameLoaded } from '../actions';
import { drawerRef } from '../lib/drawer';
import { modalRef } from '../lib/modal';
import { openSettings } from '../pages/Settings/actions';
import { loadGameFromBrowser } from '../lib/browser';
import Settings from '../pages/Settings';

const styles = {
  root: {
    position: 'relative',
  },
  headerContainer: {

  },
  footerContainer: {
    position: 'fixed',
    width: '100%',
    top: '100%',
    transform: 'translateY(-100%) translateZ(0)',
  },
  bodyContainer: {

  },
};
@connect(state => (
  {
    drawerOpen: isDrawerOpen(state),
  }
), {
  closeDrawer,
  gameLoaded,
  openSettings,
})
@injectSheet(styles)
class App extends React.Component {
  state = {
    footerHeight: 0,
  }
  componentDidMount() {
    const { gameLoaded } = this.props;
    loadGameFromBrowser()
      .then((g) => {
        if (g) gameLoaded(g);
      })
      .catch((e) => {
        console.log('Error loading game from browser: %s', e);
      });
  }
  setFooterHeight = (h) => {
    this.setState({ footerHeight: h });
  }
  render() {
    const { classes, location, openSettings } = this.props;
    return (
      <div className={classes.root}>
        <div id='drawer-root' ref={drawerRef} />
        <div id='modal-root' ref={modalRef} />
        <div className={classes.headerContainer}>
          <Header openSettings={openSettings} />
        </div>
        <div className={classes.bodyContainer} style={{ marginBottom: this.state.footerHeight }}>
          <Body location={location} />
        </div>
        <div className={classes.footerContainer}>
          <Footer onSetHeight={this.setFooterHeight} />
        </div>
        <Settings />
      </div>
    );
  }
}

export default App;
