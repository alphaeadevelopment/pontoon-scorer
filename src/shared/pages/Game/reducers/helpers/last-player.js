import findLastIndex from 'lodash.findlastindex';

export default ({ players, dealer }) => findLastIndex(players, p => p.idx !== dealer);
