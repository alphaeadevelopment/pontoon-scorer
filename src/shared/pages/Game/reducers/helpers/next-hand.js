export default ({ players, currentPlayer, currentPlayerHand }) => {
  if (players[currentPlayer].hands.length > (currentPlayerHand + 1)) {
    return currentPlayerHand + 1;
  }
  return 0;
};
