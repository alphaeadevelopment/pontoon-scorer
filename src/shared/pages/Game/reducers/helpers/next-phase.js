import { DEALER_HAND, RESULTS } from '../../../../lib/constants/game-phases';

export default ({ phase, players, currentPlayer, currentPlayerHand }) => {
  if (phase === DEALER_HAND) return RESULTS;
  if (players[currentPlayer].hands.length > (currentPlayerHand + 1)) {
    return phase;
  }
  if (currentPlayer + 1 >= players.length) {
    return phase + 1;
  }
  return phase;
};
