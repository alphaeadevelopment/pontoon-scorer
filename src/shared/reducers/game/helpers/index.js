export { default as nextPlayer } from './next-player';
export { default as nextHand } from './next-hand';
export { default as nextPhase } from './next-phase';
export { default as firstPlayer } from './first-player';
export { default as getGamePhaseAfterSetStake } from './get-game-phase-after-set-stake';
export { default as getPlayerAfterSetStake } from './get-player-after-set-stake';

