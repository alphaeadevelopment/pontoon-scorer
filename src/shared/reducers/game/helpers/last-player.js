import findLastIndex from 'lodash/findLastIndex';

export default ({ players, dealer }) => findLastIndex(players, p => p.idx !== dealer);
