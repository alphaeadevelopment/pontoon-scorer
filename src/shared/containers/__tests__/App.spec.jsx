import App from '../App';

import { createRender } from '../../../../test/src/helpers/render-helpers';

const render = createRender(App, { wrapped: true });

describe('<App />', () => {
  it('renders', () => {
    const { wrapper, getTree } = render({ classes: {} });

    expect(wrapper.exists()).toBe(true);
    expect(getTree()).toMatchSnapshot();
  });
});
