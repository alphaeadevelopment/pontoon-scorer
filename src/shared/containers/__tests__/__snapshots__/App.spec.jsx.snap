// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div>
  <div
    id="drawer-root"
  />
  <div
    id="modal-root"
  />
  <Jss(WithSizeClasses(Header)) />
  <Jss(Body)
    footerOffset={0}
  />
  <Jss(Footer)
    onSetHeight={[Function]}
  />
  <Connect(Jss(Settings)) />
  <Jss(Component)
    onCancel={[Function]}
    onLoadGame={[Function]}
    open={false}
  />
</div>
`;
